src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +136,24 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+    //this method is used to get the serverinfo object from the server
+    public ServerInfo getInfo() {
+      try (final Connection connection = this.source.connect()) {
+        Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+        if (Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE) {
+          final Uuid version = Uuid.SERIALIZER.read(connection.in());
+          return new ServerInfo(version);
+        } else {
+          // Communicate this error - the server did not respond with the type of
+          // response we expected.
+          System.out.println("Server ran into a problem retrieving the version info");
+        }
+      } catch (Exception ex) {
+          System.out.println("Connection interrupted");
+        // Communicate this error - something went wrong with the connection.
```

Comment:
Is there code missing here? Does this comment refer to the line of code above it? Generally if you're describing a line of code, place the comment above the line being described.